pkg/ipamd/ipamd.go

```diff
@@ -1102,18 +1100,66 @@ func getMinimumIPTarget() int {
 	return noMinimumIPTarget
 }
 
-func filterUnmanagedENIs(enis []awsutils.ENIMetadata) ([]awsutils.ENIMetadata, int) {
-	numFiltered := 0
+// filterUnmanagedENIs filters out ENIs marked with the "node.k8s.amazonaws.com/no_manage" tag
+func (c *IPAMContext) filterUnmanagedENIs(enis []awsutils.ENIMetadata) []awsutils.ENIMetadata {
```

Comment:
Can you add some tests for this?

Reply:
Yes, will have to rewrite most of them after these latest changes.